packages/engine/src/ui/systems/UIPanelSystem.ts

```diff
@@ -0,0 +1,70 @@
+
+import { System, SystemAttributes } from "../../ecs/classes/System";
+import { UIPanelComponent } from "../components/UIPanelComponent";
+import { Keyboard, update } from "../../assets/three-mesh-ui";
+import { createPanelComponent } from "../functions/createPanelComponent";
+import { Entity } from "../../ecs/classes/Entity";
+import { getComponent } from "../../ecs/functions/EntityFunctions";
+import { Group } from "three";
+import { Engine } from "../../ecs/classes/Engine";
+import { RaycastComponent } from "../../raycast/components/RaycastComponent";
+
+export class UIPanelSystem extends System {
+
+  panelContainer: Group;
+
+  constructor(attributes?: SystemAttributes) {
+    super(attributes);
+    this.panelContainer = new Group();
+    Engine.scene.add(this.panelContainer);
+    
+    
+    createPanelComponent({ panel: new UIPanelComponent(), raycast: new RaycastComponent() });
```

Comment:
rather than creating this in any context, we should be calling this function the client page

Reply:
this has been solved in the latest commit